core/node/node_sync/src/sync_action.rs

```diff
@@ -33,6 +33,21 @@ impl ActionQueueSender {
         Ok(())
     }
 
+    /// Pushes a single action into the queue without checking validity of the sequence.
+    ///
+    /// Useful to simulate situations where only a part of the sequence was executed on the node.
+    #[cfg(test)]
+    pub async fn push_action_unchecked(&self, action: SyncAction) -> anyhow::Result<()> {
+        self.0
+            .send(action)
+            .await
+            .map_err(|_| anyhow::anyhow!("node action processor stopped"))?;
+        QUEUE_METRICS
+            .action_queue_size
+            .set(self.0.max_capacity() - self.0.capacity());
```

Comment:
Nit: I don't think metrics are relevant for tests. Similarly, it could make sense to call `expect` on the send result.

Reply:
True, it snuck in when I copy-pasted the other method